packages/lit-html/src/test/directives/keyed_test.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {keyed} from '../../directives/keyed.js';
+import {html, render} from '../../lit-html.js';
+import {stripExpressionMarkers} from '../test-utils/strip-markers.js';
+import {assert} from '@esm-bundle/chai';
+
+suite('keyed directive', () => {
+  let container: HTMLDivElement;
+
+  setup(() => {
+    container = document.createElement('div');
+  });
+
+  test('re-renders when the key changes', () => {
+    const go = (k: any) =>
+      render(keyed(k, html`<div .foo=${k}></div>`), container);
+
+    // Initial render
+    go(1);
+    const div = container.firstElementChild;
+    assert.equal(stripExpressionMarkers(container.innerHTML), '<div></div>');
+    assert.equal((div as any).foo, 1);
+
+    // Rerendering with same key should reuse the DOM
+    go(1);
+    const div2 = container.firstElementChild;
+    assert.equal(stripExpressionMarkers(container.innerHTML), '<div></div>');
+    assert.equal((div2 as any).foo, 1);
+    assert.strictEqual(div, div2, 'X');
```

Comment:
The `X` here is extraneous, right?

Reply:
Oh yeah, it's an assertion label for debugging. Will remove.